explainer.md

```diff
@@ -1,3 +1,122 @@
+
+#### [API 1]: captureTimestamp
+
+This specification adds captureTimestamp, type of DOMHighResTimeStamp, to the RTCRtpContributingSource dictionary. This surfaces the absolute capture timestamp in the absolute capture time RTP header extension, when it is present or can be extrapolated from previously received data, for the last rendered audio or video frame. It can be used for measuring audio video synchronization performance as illustrated in the following exemplar code:
+
+    [receiverAudio, receiverVideo] = peerconnection.getReceivers();
+
+    latestCaptureTimestampsAudio = receiverAudio.getSynchronizationSources()[0].captureTimeStamp;
+    latestCaptureTimestampsVideo = receiverVideo.getSynchronizationSources()[0].captureTimeStamp;
+
+    avSynchronization = latestCaptureTimestampsAudio - latestCaptureTimestampsVideo;
+
+#### [API 2]: senderCaptureTimeOffset
+
+This specification also adds senderCaptureTimeOffset, type of DOMHighResTimeStamp, to the RTCRtpContributingSource dictionary. In this context, the sender refers to the system that directly sends RTP and RTCP packets to the receiver, and thus the sender-receiver path only represents the "last hop" in a system that involves intermediate stream regenerators.
+
+An exemplar code to use captureTimestamp and senderCaptureTimeOffset to calculate end-to-end delay follows:
+
+    receiver = peerconnection.getReceivers()[0];
+    csrc = receiver.getSynchronizationSources()[0];
+    latestCaptureTimestamps = csrc.captureTimeStamp;
+    latestSenderCaptureTimeOffset = csrc.SenderCaptureTimeOffset;
+    receiverTimestamp = csrc.timestamp;
+    
+    // Calculates sender-receiver clock offset from stats.
+    stats = peerconnection.getStats();
+    remoteOutboundRtpStats = getRequiredStats(stats, "remote-outbound-rtp");
+    remoteInboundRtpStats = getRequiredStats(stats, "remote-inbound-rtp")
+    senderReceiverTimeOffset = remoteOutboundRtpStats.timestamp - (remoteOutboundRtpStats.remoteTimestamp + remoteInboundRtpStats.roundTripTime / 2);
+    
+    // Calcuates sender-capturer clock offset.
+    captureReceiverTimeOffset = senderReceiverTimeOffset + latestSenderCaptureTimeOffset;
+    
+    receiverCaptureTimestamp = latestCaptureTimestamps + captureReceiverTimeOffset;
+    endToEndDelay = receiverTimestamp - receiverCaptureTimestamp.
+
+#### Detailed design discussion
+
+##### [Tricky design choice 1]: multiple ways to surface end-to-end delay
+
+The proposed solution is, as [API 2], to surface the sender-capture time offset as the raw data existing the absolute capture time RTP header extension. Then the calculation of end-to-end delay requires other data, as depicted in the code example in [API 2]. Alternatively, we can surface another derived quality, even just the end-to-end delay. However, for the sake of clarity and testability in the specification, it would be best to report the raw data.
+
+#### Considered alternatives
+
```

Comment:
You should probably start off this section by saying something like "We believe that given the choice of signaling protocol, the API suggested here is obvious. This section will sketch some of the other proposals for signaling protocols that have been considered and rejected over the development of this proposal."